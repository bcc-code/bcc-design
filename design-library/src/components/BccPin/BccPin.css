@layer components {
    .bcc-pin {
        @apply h-5 w-5 inline-flex items-center justify-center border rounded-full shadow-md text-xs font-semibold;
        /* Default neutral variant */
        @apply border-neutral-200 bg-neutral-50 text-neutral-900;
    }

    .bcc-pin-text {
        @apply w-auto px-2;
    }

    .bcc-pin .bcc-pin-icon {
        @apply h-4 w-4;
    }

    .bcc-pin-danger {
        @apply border-danger bg-danger text-danger;
    }
    .bcc-pin-warning {
        @apply border-warning bg-warning text-warning;
    }
    .bcc-pin-success {
        @apply border-success bg-success text-success;
    }
    .bcc-pin-info {
        @apply border-info bg-info text-info;
    }
}