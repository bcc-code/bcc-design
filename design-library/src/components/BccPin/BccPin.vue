<script setup lang="ts">
import type { Component } from "vue";

withDefaults(
  defineProps<{
    icon?: string | Component | Function;
    context?: "neutral" | "danger" | "warning" | "success" | "info";
    text?: string;
  }>(),
  {
    context: "neutral",
  }
);
</script>

<template>
  <div
    class="bcc-pin"
    :class="{
      'bcc-pin-text': !icon,
      'bcc-pin-danger': context === 'danger',
      'bcc-pin-warning': context === 'warning',
      'bcc-pin-success': context === 'success',
      'bcc-pin-info': context === 'info',
    }"
  >
    <component :is="icon" class="bcc-pin-icon" v-if="icon" />
    <span v-else-if="text">{{ text }}</span>
  </div>
</template>
