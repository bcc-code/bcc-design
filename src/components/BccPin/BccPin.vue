<script setup lang="ts">
import type { Component } from "vue";

withDefaults(
  defineProps<{
    icon?: string | Component | Function;
    variant?: "neutral" | "danger" | "warning" | "success" | "info" | "system";
  }>(),
  {
    variant: "neutral",
  }
);
</script>

<template>
  <div
    class="bcc-pin"
    :class="{
      'bcc-pin-neutral': variant === 'neutral',
      'bcc-pin-danger': variant === 'danger',
      'bcc-pin-warning': variant === 'warning',
      'bcc-pin-success': variant === 'success',
      'bcc-pin-info': variant === 'info',
      'bcc-pin-system': variant === 'system',
    }"
  >
    <component :is="icon" class="bcc-pin-icon" />
  </div>
</template>
