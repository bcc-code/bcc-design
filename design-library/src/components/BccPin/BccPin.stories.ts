import { CheckCircleIcon } from "@bcc-code/icons-vue";
import BccPin from "./BccPin.vue";

import type { Meta, setup, StoryFn } from "@storybook/vue3";

const ContextOptions = [
  "neutral",
  "danger",
  "warning",
  "success",
  "info",
  "muddy-waters",
  "mongoose",
  "brand",
];

/**
 * A companion to the badge component, to render just an icon or a number like "42" or a counter like "4/10". See the `BccBadge` component for a larger, differently styled version to be used with text.
 */
export default {
  title: "Other/BccPin",
  component: BccPin,
  argTypes: {
    context: {
      description: "Which context color to use for the styling of the pin",
      options: ContextOptions,
      control: { type: "radio" },
    },
    size: {
      description: "Size of the pin",
      options: ["sm", "base", "lg"],
      default: "base",
      control: { type: "radio" },
    },
    contrast: {
      description: "contrast",
      options: ["light", "dark"],
      control: { type: "radio" },
    },
    bordered: {
      control: { type: "boolean", default: false },
    },
    squared: {
      control: { type: "boolean", default: false },
    },
  },
} as Meta<typeof BccPin>;

const Template: StoryFn<typeof BccPin> = (args) => ({
  components: { BccPin },
  setup() {
    return { args, CheckCircleIcon };
  },
  template: `
    <BccPin v-bind="args" :icon="CheckCircleIcon" />
  `,
});

/**
 * Set the `context` prop to control how the pin looks
 */
export const Example = Template.bind({});
Example.args = {
  context: "success",
  text: "",
};
Example.parameters = {
  docs: {
    source: {
      language: "html",
      code: `
<BccPin context="success" :icon="CheckCircleIcon" />      
`,
    },
  },
};

/**
 * Set the `icon` prop to render an icon in the pin
 */
export const WithIcon: StoryFn<typeof BccPin> = () => ({
  components: { BccPin },
  setup() {
    return { CheckCircleIcon };
  },
  template: `
    <div class="flex items-start space-x-2">
      ${ContextOptions.map((o) => `<BccPin context="${o}" :icon="CheckCircleIcon" />`).join("\n")}
    </div>
  `,
});

/**
 * Set the `contrast` prop to render a pin with a light or dark contrast
 */
export const Contrast: StoryFn<typeof BccPin> = (args) => ({
  components: { BccPin },
  setup() {
    return { CheckCircleIcon, args };
  },
  template: `
    <div class="flex items-start space-x-2 mb-2">
      ${ContextOptions.map(
        (o) => `<BccPin v-bind="args" context="${o}" :icon="CheckCircleIcon" contrast="light" />`
      ).join("\n")}
    </div>
    <div class="flex items-start space-x-2">
      ${ContextOptions.map(
        (o) => `<BccPin v-bind="args" context="${o}" :icon="CheckCircleIcon" contrast="dark" />`
      ).join("\n")}
    </div>
  `,
});

/**
 * Set the `text` prop to render a text in the pin. If both `icon` and `text` are given only the `icon` is rendered
 */
export const WithText: StoryFn<typeof BccPin> = (args) => ({
  components: { BccPin },
  setup() {
    return { args };
  },
  template: `
    <div class="flex items-start space-x-2">
      <BccPin v-bind="args" context="neutral" text="1" />
      <BccPin v-bind="args" context="danger" text="2" />
      <BccPin v-bind="args" context="warning" text="10" />
      <BccPin v-bind="args" context="success" text="43" />
      <BccPin v-bind="args" context="info" text="4 / 10" />
    </div>
  `,
});
