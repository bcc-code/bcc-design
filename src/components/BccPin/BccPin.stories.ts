import BccPin from "./BccPin.vue";
import { CheckCircleIcon } from "@bcc-code/icons-vue";

import type { Meta, StoryFn } from "@storybook/vue3";

/**
 * A companion to the `Badge` component, to render just a small icon or text
 */
export default {
  title: "Components/BccPin",
  component: BccPin,
  argTypes: {
    context: {
      description: "Which context color to use for the styling of the pin",
      options: ["neutral", "danger", "warning", "success", "info"],
      control: { type: "radio" },
    },
  },
} as Meta<typeof BccPin>;

const Template: StoryFn<typeof BccPin> = (args) => ({
  components: { BccPin },
  setup() {
    return { args, CheckCircleIcon };
  },
  template: `
    <BccPin v-bind="args" :icon="CheckCircleIcon" />
  `,
});

/**
 * Set the `context` prop to control how the pin looks
 */
export const Example = Template.bind({});
Example.args = {
  context: "success",
  text: "",
};
Example.parameters = {
  docs: {
    source: {
      language: "html",
      code: `
<BccPin context="success" :icon="CheckCircleIcon" />      
`,
    },
  },
};

/**
 * Set the `icon` prop to render an icon in the pin
 */
export const WithIcon: StoryFn<typeof BccPin> = () => ({
  components: { BccPin },
  setup() {
    return { CheckCircleIcon };
  },
  template: `
    <div class="flex items-start space-x-2">
      <BccPin context="neutral" :icon="CheckCircleIcon" />
      <BccPin context="danger" :icon="CheckCircleIcon" />
      <BccPin context="warning" :icon="CheckCircleIcon" />
      <BccPin context="success" :icon="CheckCircleIcon" />
      <BccPin context="info" :icon="CheckCircleIcon" />
    </div>
  `,
});

/**
 * Set the `text` prop to render a text in the pin. If both `icon` and `text` are given only the `icon` is rendered
 */
export const WithText: StoryFn<typeof BccPin> = () => ({
  components: { BccPin },
  template: `
    <div class="flex items-start space-x-2">
      <BccPin context="neutral" text="1" />
      <BccPin context="danger" text="1" />
      <BccPin context="warning" text="1" />
      <BccPin context="success" text="1" />
      <BccPin context="info" text="1" />
    </div>
  `,
});
