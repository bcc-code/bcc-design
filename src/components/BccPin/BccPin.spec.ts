import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import BccPin from "./BccPin.vue";

describe("BccPin", () => {

});
