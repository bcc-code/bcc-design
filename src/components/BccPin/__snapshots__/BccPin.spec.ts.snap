// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BccPin > renders a pin 1`] = `
"<div class=\\"bcc-pin\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" aria-hidden=\\"true\\" viewBox=\\"0 96 960 960\\" fill=\\"currentColor\\" class=\\"bcc-pin-icon\\">
    <path d=\\"m420.522 660.782-93.457-93.456q-9.956-9.956-24.391-9.956-14.435 0-25.631 10.956-10.956 11.143-10.956 25.484 0 14.342 10.956 24.299l119.609 119.608q10.196 10.196 23.87 10.196 13.674 0 23.869-10.196l237.37-237.608q9.956-10.076 9.456-24.332-.5-14.255-10.456-24.451-10.143-10.956-24.985-10.956-14.841 0-26.037 10.956L420.522 660.782Zm59.598 321.196q-86.316 0-160.508-30.998-74.193-30.998-128.893-85.699-54.701-54.7-85.699-128.893-30.998-74.192-30.998-160.508 0-85.315 30.941-159.403 30.942-74.088 85.699-128.893 54.757-54.805 128.95-86.303 74.192-31.499 160.508-31.499 85.315 0 159.416 31.468 74.102 31.467 128.916 86.268 54.815 54.801 86.29 128.883Q886.218 490.484 886.218 576q0 86.196-31.499 160.388-31.498 74.193-86.303 128.95-54.805 54.757-128.893 85.699-74.088 30.941-159.403 30.941ZM480 576Zm-.044 337.848q144.087 0 240.99-96.741 96.902-96.741 96.902-241.065 0-144.325-96.859-241.107-96.858-96.783-240.945-96.783t-240.99 96.741q-96.902 96.741-96.902 241.065 0 144.325 96.859 241.107 96.858 96.783 240.945 96.783Z\\"></path>
  </svg></div>"
`;
